### ISSUE TITLE
Migrate to Heroku

### ISSUE THREAD

**Issue by jeancochrane** [security-force-monitor/sfm-cms#710]
We'd like to switch over to Heroku for deployments. Luckily we have some [nice guidance](https://github.com/datamade/how-to/blob/master/heroku/deploy-a-django-app.md) for doing exactly this. Some additional considerations include:

- **Solr**: We'll need to refactor the [indexing](https://github.com/security-force-monitor/sfm-cms/blob/master/search/management/commands/make_search_index.py) and [querying](https://github.com/security-force-monitor/sfm-cms/blob/ce84206da1f43fc5a68ce483993165484a09f470/search/views.py#L58-L62) code to connect via the `$SOLR_URL` environment variable instead of the settings file. We'll also need to enable [Websolr](https://elements.heroku.com/addons/websolr) in `heroku.yml` and `app.json`, test to make sure it provisions properly for deploy previews, and choose the right plan for production.
- **Translations**: Right now our translations setup relies on having persistent files on the server. We'll want to change this so that translation files are always kept up to date on version control. We have some [suggested guidance](https://github.com/datamade/how-to/blob/master/django/translation.md#keep-message-files-under-version-control) on this in our docs but we've never deployed it to production before so we should test out this workflow before we decide on it. (We've talked about [switching to a different translation service](https://github.com/security-force-monitor/sfm-cms/issues/671#issuecomment-599619112) that would make syncing with GitHub easier, so now may be the right time to reopen that conversation.)
- **Guidance for new instances**: A main motivation behind this migration is that SFM would like to be able to quickly spin up new instances. This should be pretty easy through some combination of `heroku create` and `heroku run` in the CLI but we should figure this out exactly and provide some written docs for the SFM team.
- **QA**: We'll want to do extensive QA of the new instance before cutting over DNS. In the past we've kept AWS resources running for a week or two after we cut over DNS in order to provide a backup, and I think that would be good in this case as well.

**Comment by jeancochrane** [comment]
It would probably make sense to fold this in with #708, since we'll need to clean up similar configs for all of the other instances in addition to Sahel.

**Comment by hancush** [comment]
Before I forget about it, I spec'ed out monthly hosting costs for staging and production instances of WWIC: https://docs.google.com/document/d/17AZVnd03Vh5JL_Yrs_L9bIt6WURYTtQUyoyLmL6La4k/edit?usp=sharing. It does represent an increase in monthly costs from AWS infrastructure (which has been ~$70/month, according to the console), but we've found that Heroku takes much less time to set up and maintain, which really adds up.

**Comment by hancush** [comment]
We have decided to abandon the Heroku migration. The main reasons are: 

- Websolr running Solr >= 6.6 is cost prohibitive ($549/instance/month at the time of writing), as well as manual to configure and difficult, if not impossible, to debug, as error logs are not exposed in the Websolr dashboard. This is also several versions behind the latest Solr version.
- Deployment scenarios where the app and database live on Heroku and communicate securely with a remotely hosted Solr instance would represent a net gain in complexity, undermining the dream of seamless ad hoc deployments.

If Solr were replaced with an alternative search engine, e.g., ElasticSearch, there could be a viable path forward with Heroku. The Haystack migration made that a bit easier, as switching Haystack backends is trivial, but the org and command chart code still queries Solr directly and would need to be refactored, and of course, ElasticSearch would need to be configured and administered.

Closing, for now.
